src/components/Faucet/Faucet.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState, useEffect, useMemo } from "react";
+import { TezosToolkit } from "@mavrykdynamics/taquito";
+import "./styled.css";
+
+import {
+  Network,
+  UserContext,
+  StatusContext,
+  TransactionType,
+  TestnetContext,
+} from "../../lib/Types";
+import { AddressField } from "./AddressField";
+import { AmountField } from "./AmountField";
+import { TransactionTypeSelect } from "./TransactionTypeSelect";
+import { TokenSelect } from "./TokenSelect";
+import FaucetRequestButton from "./FaucetRequestButton";
+import { MyAddress } from "./MyAddress";
+
+export type FormState = {
+  tokenAmount: string;
+  address: string;
+  transactionType: string | null;
+  selectedToken: string | null;
+  isAddressError: boolean;
+};
+
+export default function Faucet({
+  network,
+  user,
+  Tezos,
+  testnet,
+}: {
+  network: Network;
+  user: UserContext;
+  Tezos: TezosToolkit;
+  testnet: TestnetContext;
+}) {
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [status, setStatus] = useState<string>("");
+  const [statusType, setStatusType] = useState<string>("");
+  const [powWorker, setPowWorker] = useState<Worker | null>(null);
+
+  const [formState, setFormState] = useState<FormState>({
+    tokenAmount: "",
+    address: "",
+    transactionType: null,
+    selectedToken: null,
+    isAddressError: true,
+  });
+
+  const readBalances = async (): Promise<void> => {
+    try {
+      const userBalance = await Tezos.tz.getBalance(user.userAddress);
+      user.setUserBalance(userBalance.toNumber());
+    } catch (error) {
+      //console.log(error);
+    }
+  };
+
+  const statusContext: StatusContext = {
+    isLoading,
+    setLoading,
+    status,
+    setStatus,
+    statusType,
+    setStatusType,
+    powWorker,
+    setPowWorker,
+    readBalances
+  };
+
+  const isDisabledButton = useMemo(() => {
+    const {
+      transactionType,
+      tokenAmount,
+      selectedToken,
+      address,
+      isAddressError,
+    } = formState;
+    if (transactionType === TransactionType.address)
+      return (
+        !address ||
+        statusContext.isLoading ||
+        !tokenAmount ||
+        !selectedToken ||
+        isAddressError
+      );
+    if (transactionType === TransactionType.wallet)
+      return (
+        !user.userAddress ||
+        statusContext.isLoading ||
+        !tokenAmount ||
+        !selectedToken
+      );
+    return true;
+  }, [formState, statusContext, user]);
+
+  const requestAddress = useMemo(
+    () =>
+      formState.transactionType === TransactionType.address
+        ? formState.address
+        : user.userAddress,
+    [formState, user],
+  );
+
+  useEffect(() => {
+    readBalances();
+  }, [isLoading]);
+
+  return (
+    <div className="faucet-main-wrapper">
+      <div className="faucet-info-block">
```

Comment:
better to use css modules 

import styles from './myStyle.module.css'

<div className={styles.blahblah} />